### PR TITLE
Fixed out of bounds for ingredientList

### LINKED ISSUE #64: Bug in taking in 0 as a step input and only throwing error when attempting to edit step
image

### LINKED ISSUE #77: [PE-D][Tester D] Index Checking for Edit step only checks > n and not 0 or below
The team does check for index that are positively above the number of steps. However, if 0 or negative numbers are inputted by the user, the user is able to enter the description. The team however did a great job in still catching that there will be an error in viewing the recipe.

Error Catching for numbers > n:

Error Catching failing for numbers <=0:

-------------
Labels: `severity.Low` `type.FeatureFlaw`
original: shawntangy/ped#2

### LINKED ISSUE #79: [PE-D][Tester C] Strange error message when index is out of bounds
For index <= 0 

-------------
Labels: `type.FunctionalityBug` `severity.Medium`
original: honglinshang/ped#10

### LINKED ISSUE #81: [PE-D][Tester D]  Index Checking for Edit ingredient only checks > n and not 0 or below
Similar to edit step, the same minor bug occurs.

-------------
Labels: `severity.Low` `type.FeatureFlaw`
original: shawntangy/ped#3

### PR DESCRIPTION
**Pull request by LimHongYao**
Closes #64 
Fixes #77 
Fixes #79 
Fixes #81

### CODE REVIEW

**Comment by LimHongYao** [comment]
Also added proper exception class for specifying valid range

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -162,6 +162,32 @@ In the command line, the user inputs `find MaLaXiangGuo` to find a recipe from t
...
+* `RecipeList#getRecipe()` - Retrieves the `recipe` object to be edited
+* `Recipe#getStepList()` - Retrieves the `StepList` of the recipe object to be edited
+* `StepList#editStep()` - Takes in a `stepIndex` of the step to edit, and the user's input,
+then replaces the step stored at `stepIndex` with a newly created one
+
+#### Example Usage
+The example usage is based on the assumption that there currently exists at least one step in
+the recipe specified.
+
+**Step 1.** In the command line, user inputs `editstep 1` to edit the step in the first `Recipe`
```
Comment (Aung-Phone-Naing) [inline docs/DeveloperGuide.md:178]: The "Step 1" has a "." at the end that is preventing it from being bolded, may want to move it out of the formatting for bold.

**Review by Aung-Phone-Naing — CHANGES_REQUESTED** [review]
Once you have made the minor change should be good to go!

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (LimHongYao) [reply docs/DeveloperGuide.md]: removed dot

**Review by Aung-Phone-Naing — APPROVED** [review]
Okie good to go

**Pull request merged**